superset-frontend/src/components/FilterableTable/FilterableTable.tsx

```diff
@@ -321,21 +321,30 @@ export default class FilterableTable extends PureComponent<
     );
   }
 
+  // Parse any floating numbers so they'll sort correctly
+  parseFloatingNums = (value: any) => {
+    const floatValue = parseFloat(value);
+    return Number.isNaN(floatValue) ? value : parseFloat(value);
```

Comment:
Nit:

```suggestion
    return Number.isNaN(floatValue) ? value : floatValue 
```

Reply:
Whoops, I missed that one! Thank you, fixed in [`this commit`](https://github.com/apache/superset/pull/17573/commits/d1d07516e0e9e3cc7d9a54966b884803aa4158e3) 😁